pkg/operation/botanist/kubeapiserverexposure.go

```diff
@@ -79,7 +79,11 @@ func (b *Botanist) DeployKubeAPIServerSNI(ctx context.Context) error {
 }
 
 func (b *Botanist) setAPIServerServiceClusterIP(clusterIP string) {
-	b.APIServerClusterIP = clusterIP
+	if len(b.Shoot.Networks.Services.IP)*8 == 32 {
+		b.APIServerClusterIP = clusterIP
+	} else {
+		b.APIServerClusterIP = "64:ff9b::" + clusterIP
```

Comment:
Please add a short comment that this is [rfc 6052](https://datatracker.ietf.org/doc/html/rfc6052) and not some random IPv6 range.

Reply:
I changed the prefix to `64:ff9b:1::` that should be used for translation of private IPv4 addresses and added a comment.